src/cpprealm/internal/network/network_transport.cpp

```diff
@@ -78,33 +82,99 @@ namespace realm::internal {
         realm::sync::network::ReadAheadBuffer m_read_buffer;
     };
 
+    inline std::string host_from_url(const std::string& url) {
```

Comment:
You should use `<realm/util/uri.hpp>` here, the `Uri` class already has the ability to parse an uri string and give you the basic components.